projects/admin/src/app/circulation/main-request/main-request.component.ts

```diff
@@ -26,78 +28,187 @@ import { ItemsService } from '../../service/items.service';
+  ];
+
+  /** the placeholder string used on the */
+  public placeholder: string = this._translateService.instant('Please enter an item barcode.');
+  /** search text used into the search input component */
   public searchText = '';
-  public items: any[] = null;
-  private libraryPid: string;
+  /** requested items loaded */
+  public items: any[] = [];
+  /** detect if the requested items are loading */
   public isLoading = false;
-
+  /** the interval (in millis) between 2 calls of requested items (0 = no refresh) */
+  public refreshInterval = 0;
+  /** is the requested items detail should be collapsed or not */
+  public isDetailCollapsed = true;
   /** Focus attribute of the search input */
-  searchInputFocus = false;
+  public searchInputFocus = false;
+
+  /** the library pid for which load the requested items */
+  private _libraryPid: string;
+  /** the subscription for the interval refreshing */
+  private _intervalSubscription = new Subscription();
+  /** the sort criteria used */
+  private _sortCriteria = this.sortingCriteria[1].value;
 
+
+  // COMPONENT CONSTRUCTOR ===========================================================
   /** Constructor
-   * @param  userService: User Service
-   * @param  itemsService: Items Service
-   * @param  translate: Translate Service
-   * @param  toastService: Toastr Service
+   * @param _userService: User Service
+   * @param _itemsService: Items Service
+   * @param _translateService: Translate Service
+   * @param _toastService: Toastr Service
    */
   constructor(
-    private userService: UserService,
-    private itemsService: ItemsService,
-    private translate: TranslateService,
-    private toastService: ToastrService,
+    private _userService: UserService,
+    private _itemsService: ItemsService,
+    private _translateService: TranslateService,
+    private _toastService: ToastrService,
   ) {}
 
+  /** OnInit hook */
   ngOnInit() {
-    const user = this.userService.getCurrentUser();
+    const user = this._userService.getCurrentUser();
     if (user) {
-      this.libraryPid = user.getCurrentLibrary();
-      this.items = null;
+      this._libraryPid = user.getCurrentLibrary();
       this.getRequestedLoans();
+      this._enableAutoRefresh();
     }
     this.searchInputFocus = true;
   }
 
-  getRequestedLoans() {
-    this.isLoading = true;
-    this.itemsService.getRequestedLoans(this.libraryPid).subscribe(items => {
-      this.items = items;
-      this.isLoading = false;
+  /** OnDestroy hook */
+  ngOnDestroy() {
+    if (this._intervalSubscription) {
+      this._intervalSubscription.unsubscribe();
+    }
+  }
+
+  // PRIVATE FUNCTIONS ========================================================================
+  /** Enable the requested items auto-refresh behavior if needed */
+  private _enableAutoRefresh() {
+    if (this._intervalSubscription) {
+      this._intervalSubscription.unsubscribe();
+      this._intervalSubscription = new Subscription();
```

Comment:
Is it required?

Reply:
I searched a lot ! Yes it seems required :: When you unsubscribe from a subscription, it is set as 'closed' and cannot be reopened after. My solution recreate a new subscription and the process can continue